lib/prometheus_exporter/middleware.rb

```diff
@@ -39,9 +42,47 @@ def call(env)
     @client.send_json(
       type: "web",
       timings: info,
+      queue_time: queue_time,
       action: action,
       controller: controller,
       status: status
     )
   end
+
+  private
+
+  # measures the queue time (= time between receiving the request in downstream
+  # load balancer and starting request in ruby process)
+  def measure_queue_time(env)
+    start_time = queue_start(env)
+
+    return unless start_time
+
+    queue_time = request_start.to_f - start_time.to_f
+    queue_time unless queue_time.negative?
+  end
+
+  # need to use CLOCK_REALTIME, as nginx/apache write this also out as the unix timestamp
+  def request_start
+    Process.clock_gettime(Process::CLOCK_REALTIME)
+  end
+
+  QUEUE_TIME_HEADERS = %w(HTTP_X_REQUEST_START HTTP_X_QUEUE_START)
+
+  # get the content of the x-queue-start or x-request-start header
+  def queue_start(env)
+    header = QUEUE_TIME_HEADERS.detect { |h| env.key?(h) }
+    value = env[header]
+    unless value.nil? || value == ''
```

Comment:
oh I think you can return a '' back which is not handled right by measure_queue_time

Reply:
not sure what you mean here... care to elaborate?